sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java

```diff
@@ -127,11 +145,64 @@ public void describe(SensorDescriptor descriptor) {
 
   @Override
   public void execute(SensorContext context) {
+    initParsingErrorKey();
+
     for (CompatibleInputFile inputFile : wrap(fileSystem.inputFiles(mainFilesPredicate), context)) {
       XmlFile xmlFile = new XmlFile(inputFile, fileSystem);
+      try {
+        computeLinesMeasures(context, xmlFile);
+        runChecks(context, xmlFile);
+      } catch (ParseException e) {
+        processParseException(e, context, inputFile);
+      } catch (RuntimeException e) {
+        processException(e, context, inputFile);
+      }
+    }
+  }
+
+  private void initParsingErrorKey() {
+    for (Object obj : checks.all()) {
+      AbstractXmlCheck check = (AbstractXmlCheck) obj;
+      if (check instanceof ParsingErrorCheck) {
+        parsingErrorRuleKey = checks.ruleKey(check);
+        break;
+      }
+    }
+  }
+
+  private void processParseException(ParseException e, SensorContext context, CompatibleInputFile inputFile) {
+    reportAnalysisError(e, context, inputFile);
+
+    if (parsingErrorRuleKey == null) {
```

Comment:
@ivandalbosco I suggest to apply the principle of least surprise: it would be quite surprising that a line disappears from the logs when a rule is activated. Analysis logs are not always monitored by the same people who check issues in the SQ interface.